### PR TITLE
Same tweaks as before with a bit extra

### PR DESCRIPTION
**Pull request by Veldrovive**
With the fear of sounding unprofessional, don't commit a bunch of stuff and create a ton of conflicts this time, ok?

I changed it so that it will wait for school hours to come before starting the program and then I just made it so it uses time to generate the percentage instead of block.
Also I do have the gitignore there so that it doesn't take the .idea folder.

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -22,6 +22,10 @@ function getLunchInfo(lunchMenu) { // Bypassing CORS using JSONP
   }
 }
 
+function getTimeOfDay(date){
+  return date.getHours()*(60*60*1000) + date.getMinutes()*(60* 1000) + date.getSeconds()*(1000) + date.getMilliseconds();
```
Comment (HeroCC) [inline js/main.js]: What units does this return in? Seconds? Ms? Make the function name more descriptive

File: js/main.js [hunk 2]
```diff
@@ -16,18 +16,19 @@ var HttpClient = function() { // Thanks http://stackoverflow.com/a/22076667/1709
...
 }
 
 function clock(isHalfDay){
   // Thanks to http://stackoverflow.com/a/36524883/1709894 and https://www.w3schools.com/howto/howto_js_countdown.asp
+  const start = new Date();
+  start.setHours(7, 45, 0, 0);
+  const startTime = start.getTime();
   var countDownDate = new Date();
-  var time =+ countDownDate;
+  var time = +countDownDate;
```
Comment (HeroCC) [inline js/main.js:35]: Eww

File: js/main.js [hunk 3]
```diff
@@ -39,20 +44,40 @@ function clock(isHalfDay){
...
-  var seconds = Math.floor((distance % (1000 * 60)) / 1000);
-
-  document.getElementById("timer").innerHTML = hours + "h " + minutes + "m " + seconds + "s ";
-
-  if (distance < 0) {
-    document.getElementById("timer").innerHTML = "EXPIRED";
-  }
-}, 1000);
+  var timer = setInterval(function() {
+    var today = new Date();
```
Comment (HeroCC) [inline js/main.js]: This is already defined in this scope (Line 1)

**Review by HeroCC — CHANGES_REQUESTED** [review]
Small stuff this time 😉

File: js/main.js (hunk 2 above)
Comment (Veldrovive) [reply js/main.js:35]: Yea. This is actually correct. We were right that += is correct, but +date actually is a typecast. This is saying turn countDownDate into milliseconds.

**Comment by HeroCC** [comment]
Looks good! Now make codacy happy, for some reason GitHub won't let me merge if the CI is failing.

**Comment by Veldrovive** [comment]
I could pull out some of the clock function to reduce the complexity, but I want to completely restart the program by running the main function which will always result in reciprocal calling of function. What I will do for now is just force a reload of the page to run the main function that way.

**Comment by HeroCC** [comment]
Not gods approve, thanks!

**Pull request merged**